pkg/collector/corechecks/systemd/systemd.go

```diff
@@ -439,14 +463,14 @@ func getPropertyBool(properties map[string]interface{}, propertyName string) (bo
 	return propValue, nil
 }
 
-func getServiceCheckStatus(activeState string) metrics.ServiceCheckStatus {
-	switch activeState {
-	case "active":
+func getServiceCheckStatus(state string, substateMapping map[string]string) metrics.ServiceCheckStatus {
+	switch substateMapping[state] {
+	case "ok":
 		return metrics.ServiceCheckOK
-	case "inactive", "failed":
+	case "warning":
+		return metrics.ServiceCheckWarning
+	case "critical":
 		return metrics.ServiceCheckCritical
-	case "activating", "deactivating":
-		return metrics.ServiceCheckUnknown
 	}
 	return metrics.ServiceCheckUnknown
```

Comment:
Maybe we can add a `debug` log, to highlight missing mapping it think it could easy troubleshooting typo in configuration.

Reply:
Sure, added it one level higher to only log something on user provided mappings and to have more context about the unit as well.